light/store/db/db.go

```diff
@@ -289,32 +283,39 @@ func (s *dbs) Size() uint16 {
 	return s.size
 }
 
-func (s *dbs) lbKey(height int64) []byte {
-	return []byte(fmt.Sprintf("lb/%s/%020d", s.prefix, height))
+func (s *dbs) sizeKey() []byte {
+	buf := make([]byte, 0)
+	key, err := orderedcode.Append(buf, s.prefix, string(prefixSize))
+	if err != nil {
+		panic(err)
+	}
+	return key
 }
 
-var keyPattern = regexp.MustCompile(`^(lb)/([^/]*)/([0-9]+)$`)
-
-func parseKey(key []byte) (part string, prefix string, height int64, ok bool) {
-	submatch := keyPattern.FindSubmatch(key)
-	if submatch == nil {
-		return "", "", 0, false
-	}
-	part = string(submatch[1])
-	prefix = string(submatch[2])
-	height, err := strconv.ParseInt(string(submatch[3]), 10, 64)
+func (s *dbs) lbKey(height int64) []byte {
+	buf := make([]byte, 0)
+	key, err := orderedcode.Append(buf, s.prefix, string(prefixLightBlock), height)
 	if err != nil {
-		return "", "", 0, false
+		panic(err)
 	}
-	ok = true // good!
-	return
+	return key
 }
 
-func parseLbKey(key []byte) (prefix string, height int64, ok bool) {
-	var part string
-	part, prefix, height, ok = parseKey(key)
-	if part != "lb" {
-		return "", 0, false
+func (s *dbs) parseHeightFromKey(key []byte) (height int64, err error) {
```

Comment:
I think this should be called e.g. `decodeLbKey()` or something similar (`parseLbKey` is fine too), since it's the inverse of `lbKey()`.